internal/controllers/ingress_controller.go

```diff
@@ -64,7 +64,7 @@ func (irec *IngressReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 	// The object is being deleted
 	if !ingress.ObjectMeta.DeletionTimestamp.IsZero() && controllerutil.ContainsFinalizer(ingress, finalizerName) {
-		return irec.DeleteIngress(ctx, *edge, ingress)
+		return irec.DeleteIngress(ctx, edge, ingress)
```

Comment:
I did type-shuffling to get rid of the dereference here. It feels unidiomatic to deref, and there didn't seem to be a reason not to just make it a pointer throughout.

Reply:
On an internal separate pr, I was just asking about this in a general sense. Curious if you have some info and i can go do some reading on my own.

> Can you explain why you changed this? Not that its right or wrong, I just don't know best practices on when to use pointers for objects vs the object itself. Admittedly, sometimes i go with whatever makes the red line go away but I think there's some deeper reasons people have sometimes like the size of the object. If anyone has some knowledge to drop, i'd be curious

Reply:
I'll put my opinion on the general thing here, but you may find other people have slightly different opinions.

In general, if something has methods on it, it ends up _often_ being less confusing to use a pointer receiver for those methods, and if it has pointer receiver methods, I personally think it also makes sense to use the struct as a reference since I don't really like the implicit `t.foo()` reference that happens for a non-pointer struct w/ a pointer-receiver method.

The reason it's often less confusing to have a pointer receiver is the following:

https://go.dev/play/p/hRB_lcPMJmz

If you're going to have a struct that is just immutable data or where you know you're doing interior mutability right, then it's nice to use it as a non-pointer value in order to encourage the compiler to let it live on the stack, and so people know it can't be nil. In reality, the compiler will take pointer/non-pointer as more of a hint than anything else, and internally may allocate it on the stack/heap as appropriate based on escape analysis for the value.

The size of the struct also matters, as you alluded to, but you can find plenty of people micro-optimizing and writing about that online already.

Since pointers have less confusing mutation semantics, and heap-allocation is often better for sufficiently large structs, I personally find myself defaulting to using references fairly often.

-------

Note that all the talk of pointer receiver methods above is moot in this case since the `ngrokapidriver.Edge` type has no methods on it.
In this case, it was about making it consistently a pointer or not, and me defaulting to pointer since that's the easier default, especially since it was already a pointer at the place I removed the deref.